security/ThreatModelling/ThreatModelling.md

```diff
@@ -0,0 +1,93 @@
+> 2.	If you don’t know what to protect, how do you know you’re protecting it?
+> 3.	If you are not protecting it, the dragon wins. 
+
+## What is Threat Modelling
+
+At its most basic level; Threat Modelling is identifying and prioritising potential attacks, threats, and vulnerabilities within your assets. This is done by identifying assets (using methods such as data flow diagrams) to help discuss scenarios, "attack surface" areas, and countermeasures.  
+
+There are different _approaches_ and _methodologies_ to threat modelling that help support different operational functions. "_Approaches_" help support the "_Methodology_" chosen, and neither are used in isolation.  
+
+### Approaches
+
+- Asset based – Best used to describe assets when working with a multi-disciplinary team.
+- Attacker based – Best used when working with experienced security team members.
+- Application based – Best used when working in/with a team made up mostly of developers.
+
+### Methodologies
+
+- __PASTA__ – Process Attack Simulation and Threat Analysis
+- __STRIDE__ – Spoofing, Tampering, Repudiation, Information Disclosure, Denial of Service, Elevation of Privilege
+- __OCTAVE__ – Operationally Critical Threat, Asset and Vulnerability Evaluation
+- __TRIKE__ – Risk based security auditing from top level down.
+- __VAST__ – Visual, Agile, Simple Threat modelling
+
+## How do we perform threat modelling
+
+Threat modelling should be performed: 
+
+ - At the beginning of a project - (to assist with understanding potential attack vectors, and allow for designs to be changed before anything is created),
+ - Whenever there is a change to the _attack surface_ of the application.
+
+Wherever possible, a **_Data Flow Diagram_** can be created to help visualise and define the assets that are apart of the application, so that a conversation can be started about attack vectors of the different areas. The data flow diagram should be a growing diagram that is maintained in tandem with the project.
+
+The tooling of choice to support this is the **_Microsoft Threat Modelling Tool_**. Utilising the data flow diagram within the tooling, it can analyse and support the STRIDE method of giving potential attack vectors to your application. This tooling should be a compliment for the _conversation_ however, rather than a replacement for it. 
+
+To start the conversation for a Threat Modelling session, the following will need to be understood so that the session can focus on what is required from the threat modelling:
+
+- __Scope__ – What is in scope for this session?
+- __Stakeholders__ – Who has input into this system design and needs to be involved in this modelling.
+- __Risk management__ – Who will support you in assessing and devising a plan for any risks that are found.
+
+Depending on the stage of the project, facilitation can be handled in two ways:
+
+- Early project – Facilitated by Security personnel (Principals, Lead Tech, ITSO)
```

Comment:
Are we saying that a threat modelling session at the start of a project needs to have Principals, leads, and ITSO in it? Usually how it works at the moment the team does the threat modelling session among themselves and then meets with ITSO to discuss the project and what had been identified and mitigations.

Reply:
To bring our conversation here, the ideal would be what is laid out in this document as you can build up your documentation from the early stages. It also allows for intervention at the design stage when looking from the outside in on the project. Obviously an ideal and something we can work towards.